app/Validators/CreateDefinitionValidator.ts

```diff
@@ -0,0 +1,52 @@
+import { schema, rules, CustomMessages } from '@ioc:Adonis/Core/Validator'
+import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
+
+export default class CreateDefinitionValidator {
+  constructor(protected ctx: HttpContextContract) {}
+
+  /*
+   * Define schema to validate the "shape", "type", "formatting" and "integrity" of data.
+   *
+   * For example:
+   * 1. The username must be of data type string. But then also, it should
+   *    not contain special characters or numbers.
+   *    ```
+   *     schema.string({}, [ rules.alpha() ])
+   *    ```
+   *
+   * 2. The email must be of data type string, formatted as a valid
+   *    email. But also, not used by any other user.
+   *    ```
+   *     schema.string({}, [
+   *       rules.email(),
+   *       rules.unique({ table: 'users', column: 'email' }),
+   *     ])
+   *    ```
+   */
+  public schema = schema.create({
+    user_id: schema.number([rules.required()]),
+    term: schema.string([rules.maxLength(255), rules.required()]),
+    definition: schema.string([rules.minLength(10), rules.required()]),
+    category_id: schema.number([rules.required()]),
```

Comment:
no need to add rules.required() again, schema.number() makes property is required by default

Reply:
thanks for correction i will remove it